### ISSUE TITLE
Prepare the wiki for osu!(lazer)

### ISSUE THREAD

**Issue by TicClick** [ppy/osu-wiki#7139]
# the issue

most of the articles listed on the wiki only describe how the current stable build functions. this goes for things such as gameplay mechanics or chat ([osu!supporter](https://osu.ppy.sh/wiki/en/osu%21supporter) colour for instance).

as lazer may (and will) differ significantly from stable, at least with respect to certain features, we need to figure out how to handle the wiki update in advance.

# related

- https://github.com/ppy/osu-wiki/issues/6590
- https://github.com/ppy/osu-wiki/issues/4686
- user count: https://status.ppy.sh/

**Comment by TicClick** [comment]
a couple ideas from myself:

- a separate section for every such article. quick & cheap, but will definitely disrupt the flow of almost any article
- a separate namespace or a query string for all articles, such as `https://osu.ppy.sh/wiki/lazer/en/Main_Page` or `https://osu.ppy.sh/wiki/en/Main_Page?release=lazer`
- a separate **subarticle**, such as `https://osu.ppy.sh/wiki/en/Beatmap/Difficulty/Lazer` (imo the worst solution, because it will nearly double the amount of articles and likely make editing harder)

**Comment by peppy** [comment]
I think it should be done in a way that we can switch the primary source of information when the majority of players are using lazer, with the old client articles being moved to `(Legacy)` or similar.

**Comment by Walavouchey** [comment]
> a separate namespace or a query string for all articles, such as `https://osu.ppy.sh/wiki/lazer/en/Main_Page`

I think this could be a very elegant solution, but it doesn't necessarily need web support.

In practice, this would be a separate folder in the repo (`lazer` and then later `legacy`) with a mirrored file structure for the articles there.

- If the only difference is images, we can store them in the new folder (e.g. `/wiki/lazer/Client/Beatmap_editor/Beat_Snap_Divisor/img/*`) if we want to prepare them beforehand.
- If there are partial differences, make a stub explaining the differences with a hatnote pointing to it from the original article. After switching, the now "legacy" stub would explain differences in stable.
- If the whole article is stable-specific (e.g. interface articles), the lazer-specific version would just be a complete article.
- If something exists only in lazer, they just get their own article (game modifier articles for instance, with `Game_modifier` either having `(lazer only)` in the listing or having a new version of itself).

I don't think there's a need for a lazer-specific main page

**Comment by peppy** [comment]
Another options would be to maintain a `master` and `master-lazer` branch. Would avoid the need for duplicating images etc. In a future we could rename `master-lazer` to `master` and `master` to `master-legacy`.

@nanaya curious on your thoughts on this topic

**Comment by nanaya** [comment]
Separate section for where possible/makes sense and dedicated subarticle (or more like with `(lazer)` appended at the title/directory instead of subdirectory) for more involved pages (with link to each other at the top if applicable or mention it being lazer-only or legacy/stable-only) maybe? And once the main version is switched, the section and subarticle can also be switched.

For those which only differ in image, unless it looks very different, updating to lazer screenshot right away is probably fine? Maybe with a note about slight difference in version and/or link to legacy version of the image?

Or if access to other version is not needed on the web then sure the branch method can also be used 🤷

**Comment by Walavouchey** [comment]
> Another options would be to maintain a `master` and `master-lazer` branch. Would avoid the need for duplicating images etc. In a future we could rename `master-lazer` to `master` and `master` to `master-legacy`.

Once the switch happens, I don't think we should be removing stable-specific articles for the time being, if that's what this is implying.

I can get behind having a `master-lazer` branch in lieu of a different folder like I suggested, i.e. legacy articles are kept in the new branch and labelled as such while lazer-specific articles are put in place. This way, we only need to merge `master-lazer` into `master` to make the switch.

> For those which only differ in image, unless it looks very different, updating to lazer screenshot right away is probably fine?

We could, but it would make the wiki very inconsistent for the likely long interim of preparing lazer articles. Doing it all in one fell swoop, either with a new branch or a temporary folder, sounds like a better option to me.

**Comment by cl8n** [comment]
> a separate **subarticle**, such as `https://osu.ppy.sh/wiki/en/Beatmap/Difficulty/Lazer` (imo the worst solution, because it will nearly double the amount of articles and likely make editing harder)

a wiki ive worked on about Star Wars did exactly this for "Canon" and "Legends" (basically two different versions of the story). articles that span both have a switcher button at the top that links to the other one, and the contents of those articles usually have significant overlap. it can be tedious to edit both at once sometimes, but the experience as a reader is good because you're given complete information abt what ur reading, and don't have to read around any info from the other version.

with freedom of implementation here I think we can get those same benefits but avoid having to duplicate edits. for articles that would be totally different, e.g. Interface, using separate articles is fine. for articles with lots of overlap, there could be some syntax to pick between two sets of content depending on which version you are viewing, like
```markdown
+++
normal content (for lazer)
===
legacy content (for stable)
+++
```
(symbols chosen at random for example)

then, this new block can be used where the article differs between versions, and the rest of the article can be written as normal.

---

about maintaining separate branches and eventually swapping the main one, I don't think that is a good option; if done right there should be no reason that content can't appear on the site as it's added. if writing about the new client is somehow obscuring writing about the old, then that is a problem regardless of what strategy is decided here.

**Comment by peppy** [comment]
> Once the switch happens, I don't think we should be removing stable-specific articles for the time being, if that's what this is implying.

Not quite, but it would mean we'd have to continue managing two branches into the future, which may be suboptimal. Might be best to go with one of the other solutions.

**Comment by Soetch** [comment]
Hey, 
I'm new on this project, wanted to say that if this article needs a french translation, I can translate it natively.
The legacy/normal content organization seems easy to understand and I would be happy to be assigned to french translation/writing issues.

**Comment by Soetch** [comment]
> >Once the switch happens, I don't think we should be removing stable-specific articles for the time being, if that's what this is implying.

>Not quite, but it would mean we'd have to continue managing two branches into the future, which may be suboptimal. Might be best to go with one of the other solutions.

We should maybe give an osu! support until lazer is 75% populated (like, used).

**Comment by Gamingzone123** [comment]
my two cents worth is (somewhat similar to what cl8n suggested but without the need to implement the toggle solution, or potentially as an intermediary step), use the separate sections for each article and just copy/paste swap the order and rename once lazer becomes the main stable branch to change what is first seen by a reader (this could potentially be done automatically with a script?) which might go something like:


**Before the switch:**

Stable branch
-----------------
_stable article content_

-----------------
Lazer
-----------------
_new/amended article content_

***Becomes***

**after the switch:**

Stable Branch
-----------------
_newer article content_

-----------------
Legacy/Fallback
-----------------
_old article content_

**Comment by Walavouchey** [comment]
After [discussion on Discord](https://discord.com/channels/188630481301012481/218677502141399041/1069749583644733550) 
 with @TicClick and @cl8n we have agreed to have adjacent lazer versions of the articles similar to [nanaya's suggestion](https://github.com/ppy/osu-wiki/issues/7139#issuecomment-1091100597), appending ` (lazer)` to the duplicate and having them link to each other at the top.

**Comment by TicClick** [comment]
yup. now all we need is a list of such articles

**Comment by CloneWith** [comment]
# A brief article & media list

Here is the list of articles and assets to be updated / added I have found out. There may be something missing, but I'll add them as long as I can.

## Articles

There are quite some articles significantly make the difference.

### Beatmapping

- Grid snapping: ways to switch grid size
- New combo: ways to trigger
- Beatmapping: ways to quickly timing

### Client

- Beatmap editor
  - Main article: section change & interface update
  - Beat snap devisor: according to new interface changes
  - Compose, Design, Timing, (Verify to be added?): complete change
  - Menu: content update
  - Test mode
  - Timelines
  - Re~lengths, Resnap, SB load: No corresponding menu/widget entry
- Installation: Switch installation methods
- Interface
  - Main article
  - Chat console
    - Main article: interface change (channel management, newly added PMs; Extended console separated)
    - Highlight
  - Multiplayer: interface change (like changed team modes, beatmap queue, timer buttons, spectating)
    - Consider adding playlist mode?
  - Visual settings: interface change
- Ketboard shortcuts: *change unsure*
- Menu tips (do we have it?)
- Options
  - Main article: content change
  - Keyboard bindings
  - Offset Wizard
  - Add other articles
- Program files: Export locations of skins, beatmaps, replays, etc.
- Release stream: switch

### FAQ

This article is a big work and may need advice from the community.

### Game mode

- osu!taiko: slight rule change (drumrolls)

### Gameplay

- Game modifier: Newly added categories, mods and preset function. Mod icons also needs updating
- Grade: criteria
- Judgement
- Kiai time: *behavior change unsure*
- Replay: interface change
- Spectating: procedure & interface change

### Offset

## Media

Medias to be updated appear to be mostly images. Along with those images in articles above, we also have:

- Bancho (server)
  - Connection warning (`connection-warning.png`)
- Beatmapping
  - Timing -> Timeline (`complex-timing.png`)
- A lot of interface screenshots imo

**Comment by CloneWith** [comment]
There are some articles with `History` and `Trivia` sections, how should we deal with them?

I thought leaving them in stable articles is kind of inappropriate.

**Comment by Gamingzone123** [comment]
> There are some articles with `History` and `Trivia` sections, how should we deal with them?
> 
> I thought leaving them in stable articles is kind of inappropriate.

maybe move between stable (to be fallback) and lazer (to be stable) or both depending on relevance?
